js/src/creatives/companion.js

```diff
@@ -38,15 +40,27 @@ COMPANION.parse = function (companionAds) {
       if (!FW.imagePattern.test(creativeType)) {
         continue;
       }
-      let width = companion.getAttribute('width');
-      // width attribute is required
+
+      let width = companion.getAttribute('width'); // width attribute is required
+
+      let height = companion.getAttribute('height'); // height attribute is also required
+
+      const staticResourceUrl = FW.getNodeValue(staticResource[0], true);
+      if (staticResourceUrl === null) {
+        continue;
+      }
+
+      this.companionForEndCard.push({
+        width: width ? width: 0,
+        height: height ? height: 0,
+        imageUrl: staticResourceUrl
```

Comment:
looks to be a dublicate of ^ code, can we generalise & import?

Reply:
This is not a duplicate. rmp-vast.js is generated after build.